contracts/axelar-interchain-token-service/src/test.rs

```diff
@@ -31,59 +29,145 @@ fn initialize() {
     let chain = String::from_str(&env, "chain");
     let trusted_address = String::from_str(&env, "trusted_address");
 
-    client.mock_auths(&[MockAuth {
-        address: &not_owner,
-        invoke: &MockAuthInvoke {
-            contract: &contract_id,
-            fn_name: "set_trusted_address",
-            args: (chain.clone(), trusted_address.clone()).into_val(&env),
-            sub_invokes: &[],
-        },
-    }]).set_trusted_address(&chain, &trusted_address);
+    client
+        .mock_auths(&[MockAuth {
+            address: &owner,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "set_trusted_address",
+                args: (chain.clone(), trusted_address.clone()).into_val(&env),
+                sub_invokes: &[],
+            },
+        }])
+        .set_trusted_address(&chain, &trusted_address);
+
+    assert_last_emitted_event(
+        &env,
+        &contract_id,
+        (
+            Symbol::new(&env, "trusted_address_set"),
+            chain.clone(),
+            trusted_address.clone(),
+        ),
+        (),
+    );
 
+    assert_eq!(client.trusted_address(&chain), Some(trusted_address));
 }
 
 #[test]
-fn set_trusted_address() {
+#[should_panic(expected = "HostError: Error(Auth, InvalidAction)")]
+fn set_trusted_address_fails_if_not_owner() {
     let (env, contract_id, client) = setup_env();
     let owner = Address::generate(&env);
 
     initialize_its(&env, &client, owner.clone());
 
+    let not_owner = Address::generate(&env);
     let chain = String::from_str(&env, "chain");
     let trusted_address = String::from_str(&env, "trusted_address");
-    
-    client.mock_auths(&[MockAuth {
-        address: &owner,
-        invoke: &MockAuthInvoke {
-            contract: &contract_id,
-            fn_name: "set_trusted_address",
-            args: (chain.clone(), trusted_address.clone()).into_val(&env),
-            sub_invokes: &[],
-        },
-    }]).set_trusted_address(&chain, &trusted_address);
-
-    assert_eq!(
-        client.is_trusted_address(&chain, &trusted_address),
-        true
+
+    client
+        .mock_auths(&[MockAuth {
+            address: &not_owner,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "set_trusted_address",
+                args: (chain.clone(), trusted_address.clone()).into_val(&env),
+                sub_invokes: &[],
+            },
+        }])
+        .set_trusted_address(&chain, &trusted_address);
+}
+
+#[test]
+fn set_trusted_address_fails_if_already_set() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+
+    initialize_its(&env, &client, owner.clone());
+
+    let chain = String::from_str(&env, "chain");
+    let trusted_address = String::from_str(&env, "trusted_address");
+    let another_trusted_address = String::from_str(&env, "another_trusted_address");
+
+    client
+        .mock_auths(&[MockAuth {
```

Comment:
This and the following test isn't checking for auth, so better to just mock all auths `env.mock_all_auths();`